In [43]:
import pandas as pd
import numpy as np

In [44]:
actor = pd.read_csv(r"C:\Users\HP\Desktop\Data Sets\bollywood_movie\BollywoodActorRanking.csv")
actor

actorId         actorName  movieCount  ratingSum  normalizedMovieRank  \
0        373        Aamir Khan          11       1170              9.44862   
1        374    Shah Rukh Khan          23       2000              7.55009   
2        375       Salman Khan          36       2340              5.40217   
3        376      Katrina Kaif          17       1640              8.48082   
4        377  Deepika Padukone          16       1080              5.64674   
..       ...               ...         ...        ...                  ...   
296      669       Achint Kaur           4         80              1.00000   
297      670   Yuvika Chaudhry           5        100              1.00000   
298      671      Pankaj Berry           5        100              1.00000   
299      672      Varun Dhawan           0          0                  NaN   
300      673     Nargis Fakhri           0          0                  NaN   

     googleHits  normalizedGoogleRank  normalizedRating  
0       2460000               7.34283          10.00000  
1       2670000               7.88458           9.22674  
2       3490000              10.00000           9.20821  
3       2120000               6.46570           8.94852  
4       3000000               8.73591           8.62716  
..          ...                   ...               ...  
296           0                   NaN           1.00000  
297           0                   NaN           1.00000  
298           0                   NaN           1.00000  
299           0                   NaN           4.74000  
300           0                   NaN           4.80000  

[301 rows x 8 columns]

There are 301 rows and 8 columns

In [45]:
actor.describe()

actorId  movieCount    ratingSum  normalizedMovieRank    googleHits  \
count  301.000000  301.000000   301.000000           299.000000  3.010000e+02   
mean   523.000000   10.232558   419.136213             2.768970  3.741886e+05   
std     87.035433    8.508764   446.972689             1.626970  6.647169e+05   
min    373.000000    0.000000     0.000000             1.000000  0.000000e+00   
25%    448.000000    5.000000   140.000000             1.489130  0.000000e+00   
50%    523.000000    7.000000   260.000000             2.369570  7.640000e+04   
75%    598.000000   12.000000   500.000000             3.621045  3.850000e+05   
max    673.000000   54.000000  2950.000000            10.000000  3.490000e+06   

       normalizedGoogleRank  normalizedRating  
count            189.000000        301.000000  
mean               2.533937          2.933339  
std                1.951501          1.747883  
min                1.000000          1.000000  
25%                1.242680          1.709770  
50%                1.638930          2.493970  
75%                2.967520          3.559640  
max               10.000000         10.000000

* There might be null values in normalizedMovieRank and normalizedGoogleRank
* Values in movieCount ranges from 0 to 54, indicating the highest movie an actor has done might be 54. Minimum value 0 could be an error
* actorId ranges from 373 to 673
* normalizedMovieRank, normalizedGoogleRank and	normalizedRating has values ranging from 1 to 10

In [46]:
actor.describe(include = object).transpose()

count unique         top freq
actorName   301    301  Aamir Khan    1

In [47]:
actor.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 301 entries, 0 to 300
Data columns (total 8 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   actorId               301 non-null    int64  
 1   actorName             301 non-null    object 
 2   movieCount            301 non-null    int64  
 3   ratingSum             301 non-null    int64  
 4   normalizedMovieRank   299 non-null    float64
 5   googleHits            301 non-null    int64  
 6   normalizedGoogleRank  189 non-null    float64
 7   normalizedRating      301 non-null    float64
dtypes: float64(3), int64(4), object(1)
memory usage: 18.9+ KB


There are 3 columns with float data type, 4 with int data type and 1 column with object data type

In [48]:
actor.isnull().sum()

actorId                   0
actorName                 0
movieCount                0
ratingSum                 0
normalizedMovieRank       2
googleHits                0
normalizedGoogleRank    112
normalizedRating          0
dtype: int64

There are 2 values in normalizedMovieRank with null values and 112 in normalizedGoogleRank

In [49]:
actor['normalizedGoogleRank'].fillna(actor['normalizedGoogleRank'].mode()[0], inplace = True)

In [50]:
actor.dropna(inplace = True)

In [51]:
director = pd.read_csv(r"C:\Users\HP\Desktop\Data Sets\bollywood_movie\BollywoodDirectorRanking.csv")
director

directorId     directorName  movieCount  ratingSum  normalizedMovieRank  \
0             1  Rajkumar Hirani           3        440             10.00000   
1             2       Farah Khan           4        440              7.39474   
2             3      Karan Johar           5        430              5.68947   
3             4       Sajid Khan           4        350              5.79605   
4             5     Rohit Shetty          10        950              6.32895   
..          ...              ...         ...        ...                  ...   
113         114     Shyam Ramsay           3         60              1.00000   
114         115      Jag Mundhra           3         60              1.00000   
115         116    Kalpana Lajmi           3         60              1.00000   
116         117    Ajay Chandhok           4         80              1.00000   
117         118      Puneet Sira           3         60              1.00000   

     googleHits  normalizedGoogleRank  normalizedRating  
0        146000               2.07796          10.00000  
1       1060000               8.88317           8.78473  
2       1050000               8.80872           7.10460  
3        905000               7.72912           6.99780  
4        437000               4.24461           6.83761  
..          ...                   ...               ...  
113        9930               1.06485           1.01149  
114        6710               1.04088           1.00681  
115        4380               1.02353           1.00342  
116        2420               1.00893           1.00057  
117        2030               1.00603           1.00000  

[118 rows x 8 columns]

There are 118 rows and 8 columns

In [52]:
director.describe()

directorId  movieCount   ratingSum  normalizedMovieRank    googleHits  \
count  118.000000  118.000000   118.00000           118.000000  1.180000e+02   
mean    59.500000    4.957627   220.00000             2.601887  2.176836e+05   
std     34.207699    2.799249   182.05381             1.588392  3.322251e+05   
min      1.000000    3.000000    60.00000             1.000000  0.000000e+00   
25%     30.250000    3.000000    92.50000             1.397560  1.117500e+04   
50%     59.500000    4.000000   145.00000             2.350000  4.855000e+04   
75%     88.750000    6.000000   270.00000             3.403948  2.687500e+05   
max    118.000000   19.000000  1050.00000            10.000000  1.210000e+06   

       normalizedGoogleRank  normalizedRating  
count            118.000000        118.000000  
mean               2.594890          2.874995  
std                2.487982          1.779069  
min                0.000000          1.000000  
25%                1.074120          1.452045  
50%                1.352400          2.483650  
75%                2.991902          3.797115  
max               10.000000         10.000000

In [53]:
director.describe(include = object).transpose()

count unique              top freq
directorName   117    117  Rajkumar Hirani    1

In [54]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 117 entries, 0 to 117
Data columns (total 8 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   directorId            117 non-null    int64  
 1   directorName          117 non-null    object 
 2   movieCount            117 non-null    int64  
 3   ratingSum             117 non-null    int64  
 4   normalizedMovieRank   117 non-null    float64
 5   googleHits            117 non-null    int64  
 6   normalizedGoogleRank  117 non-null    float64
 7   normalizedRating      117 non-null    float64
dtypes: float64(3), int64(4), object(1)
memory usage: 8.2+ KB


In [55]:
director.isnull().sum()

directorId              0
directorName            1
movieCount              0
ratingSum               0
normalizedMovieRank     0
googleHits              0
normalizedGoogleRank    0
normalizedRating        0
dtype: int64

In [56]:
director.dropna(inplace = True)

In [57]:
movie = pd.read_csv(r"C:\Users\HP\Desktop\Data Sets\bollywood_movie\BollywoodMovieDetail.csv")
movie

imdbId                              title  releaseYear releaseDate  \
0     tt0118578                             Albela         2001   20-Apr-01   
1     tt0169102  Lagaan: Once Upon a Time in India         2001   08-May-02   
2     tt0187279           Meri Biwi Ka Jawab Nahin         2004   02-Jul-04   
3     tt0222024             Hum Tumhare Hain Sanam         2002   24-May-02   
4     tt0227194                         One 2 Ka 4         2001   30-Mar-01   
...         ...                                ...          ...         ...   
1279  tt4121522               Shuruaat Ka Interval         2014   15-Aug-14   
1280  tt4187650                   Trip to Bhangarh         2014   28-Aug-14   
1281  tt4190220                           Zed Plus         2014   28-Nov-14   
1282  tt4219300                    Gollu aur Pappu         2014   21-Nov-14   
1283  tt4228746                                Zid         2014   28-Nov-14   

                            genre  \
0                         Romance   
1     Adventure | Drama | Musical   
2                 Action | Comedy   
3                 Drama | Romance   
4         Action | Comedy | Drama   
...                           ...   
1279                          NaN   
1280                       Horror   
1281                          NaN   
1282                       Comedy   
1283                     Thriller   

                                                writers  \
0     Honey Irani (screenplay) | Honey Irani (story)...   
1     Ashutosh Gowariker (story) | Ashutosh Gowarike...   
2                                                   NaN   
3     K.S. Adiyaman | Arun Kumar (assistant dialogue...   
4     Sanjay Chhel | Raaj Kumar Dahima (screenplay) ...   
...                                                 ...   
1279  Aarti S. Bagdi | Amrit Raj Gupta | Krishan Hoo...   
1280                                                NaN   
1281  Chandra Prakash Dwivedi (screen play and dialo...   
1282                                                NaN   
1283                   Vivek Agnihotri | Rohit Malhotra   

                                                 actors  \
0     Govinda | Aishwarya Rai Bachchan | Jackie Shro...   
1     Aamir Khan | Gracy Singh | Rachel Shelley | Pa...   
2     Akshay Kumar | Sridevi | Gulshan Grover | Laxm...   
3     Shah Rukh Khan | Madhuri Dixit | Salman Khan |...   
4     Shah Rukh Khan | Juhi Chawla | Jackie Shroff |...   
...                                                 ...   
1279  Kaivalya Chheda | Rajsekhar Aningi | Avidant B...   
1280  Manish Choudhary | Suzanna Mukherjee | Poonam ...   
1281  Adil Hussain | Ekavali Khanna | Kulbhushan Kha...   
1282  Vir Das | Sandeepa Dhar | Rahul Handa | Dimple...   
1283  Shradha Das | Mohan Kapoor | Mannara | Karanvi...   

                                              directors  sequel  hitFlop  
0                                         Deepak Sareen     0.0        2  
1                                    Ashutosh Gowariker     0.0        6  
2                          Pankaj Parashar | S.M. Iqbal     0.0        1  
3                                         K.S. Adiyaman     0.0        4  
4                                     Shashilal K. Nair     0.0        1  
...                                                 ...     ...      ...  
1279  Aarti S. Bagdi | Amrit Raj Gupta | Krishan Hoo...     0.0        1  
1280                                     Jitendra Pawar     0.0        1  
1281                            Chandra Prakash Dwivedi     0.0        1  
1282                                     Kabir Sadanand     0.0        1  
1283                     Rohit Malhotra | Anubhav Sinha     0.0        2  

[1284 rows x 10 columns]

There are 1284 rows and 10 columns

In [58]:
movie.describe()

releaseYear       sequel      hitFlop
count  1284.000000  1281.000000  1284.000000
mean   2007.989097     0.035129     2.144081
std       4.005243     0.188371     1.812820
min    2001.000000     0.000000     1.000000
25%    2004.000000     0.000000     1.000000
50%    2008.000000     0.000000     1.000000
75%    2011.000000     0.000000     2.000000
max    2014.000000     2.000000     9.000000

In [59]:
movie.describe(include = object).transpose()

count unique                                                top  \
imdbId       1284   1284                                          tt0118578   
title        1284   1284                                             Albela   
releaseDate  1231    710                                          02-Nov-12   
genre        1282    215                                              Drama   
writers      1165   1109                                       Dilip Shukla   
actors       1281   1281  Govinda | Aishwarya Rai Bachchan | Jackie Shro...   
directors    1280    731                                       Priyadarshan   

            freq  
imdbId         1  
title          1  
releaseDate    6  
genre        113  
writers        6  
actors         1  
directors     18

In [60]:
movie.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1284 entries, 0 to 1283
Data columns (total 10 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   imdbId       1284 non-null   object 
 1   title        1284 non-null   object 
 2   releaseYear  1284 non-null   int64  
 3   releaseDate  1231 non-null   object 
 4   genre        1282 non-null   object 
 5   writers      1165 non-null   object 
 6   actors       1281 non-null   object 
 7   directors    1280 non-null   object 
 8   sequel       1281 non-null   float64
 9   hitFlop      1284 non-null   int64  
dtypes: float64(1), int64(2), object(7)
memory usage: 100.4+ KB


In [61]:
movie.isna().sum()

imdbId           0
title            0
releaseYear      0
releaseDate     53
genre            2
writers        119
actors           3
directors        4
sequel           3
hitFlop          0
dtype: int64

In [62]:
# Boolean indexing
row_with_null_value_actors = movie[movie['actors'].isna()]
row_with_null_value_actors

imdbId                         title  releaseYear releaseDate  \
601  tt1169841             Return of Hanuman         2007   28-Dec-07   
797  tt1587388  Yeh Hai Malegaon Ka Superman         2009   01-Nov-09   
835  tt1633171                     Bird Idol         2010   23-Apr-10   

                              genre                 writers actors  \
601  Animation | Action | Adventure          Anurag Kashyap    NaN   
797                 Comedy | Family                     NaN    NaN   
835                       Animation  Dev Goel | Jyotin Goel    NaN   

          directors  sequel  hitFlop  
601  Anurag Kashyap     1.0        1  
797    Nasir Shaikh     0.0        1  
835     Jyotin Goel     0.0        1

Lets remove the rows with null values

In [63]:
movie.dropna(inplace = True)

In [64]:
movie

imdbId                              title  releaseYear releaseDate  \
0     tt0118578                             Albela         2001   20-Apr-01   
1     tt0169102  Lagaan: Once Upon a Time in India         2001   08-May-02   
3     tt0222024             Hum Tumhare Hain Sanam         2002   24-May-02   
4     tt0227194                         One 2 Ka 4         2001   30-Mar-01   
5     tt0238936                             Devdas         2002   12-Jul-02   
...         ...                                ...          ...         ...   
1275  tt3893476                              Spark         2014   10-Oct-14   
1276  tt4010306                          Jigariyaa         2014   10-Oct-14   
1277  tt4045666                Mumbai Delhi Mumbai         2014   05-Dec-14   
1278  tt4119196                  Love U Crazy Girl         2014   09-May-14   
1283  tt4228746                                Zid         2014   28-Nov-14   

                            genre  \
0                         Romance   
1     Adventure | Drama | Musical   
3                 Drama | Romance   
4         Action | Comedy | Drama   
5               Musical | Romance   
...                           ...   
1275                       Action   
1276                        Drama   
1277             Comedy | Romance   
1278                      Mystery   
1283                     Thriller   

                                                writers  \
0     Honey Irani (screenplay) | Honey Irani (story)...   
1     Ashutosh Gowariker (story) | Ashutosh Gowarike...   
3     K.S. Adiyaman | Arun Kumar (assistant dialogue...   
4     Sanjay Chhel | Raaj Kumar Dahima (screenplay) ...   
5     Saratchandra Chatterjee (based on the novel by...   
...                                                 ...   
1275                                 V.K. Singh (story)   
1276       Vinod Bachchan | Apratim Khare | Raj Purohit   
1277      Manu Rishi Chadda (dialogue) | Satish Rajwade   
1278                                    Bhabhani Prasad   
1283                   Vivek Agnihotri | Rohit Malhotra   

                                                 actors  \
0     Govinda | Aishwarya Rai Bachchan | Jackie Shro...   
1     Aamir Khan | Gracy Singh | Rachel Shelley | Pa...   
3     Shah Rukh Khan | Madhuri Dixit | Salman Khan |...   
4     Shah Rukh Khan | Juhi Chawla | Jackie Shroff |...   
5     Shah Rukh Khan | Madhuri Dixit | Aishwarya Rai...   
...                                                 ...   
1275  Rati Agnihotri | Mansha Bahl | Rajneesh Duggal...   
1276  Deepak Chadha | Harshvardhan Deo | Sneha Deori...   
1277                          Shiv Pandit | Piaa Bajpai   
1278  Mohan Joshi | Shakti Kapoor | Nitika Sharma | ...   
1283  Shradha Das | Mohan Kapoor | Mannara | Karanvi...   

                           directors  sequel  hitFlop  
0                      Deepak Sareen     0.0        2  
1                 Ashutosh Gowariker     0.0        6  
3                      K.S. Adiyaman     0.0        4  
4                  Shashilal K. Nair     0.0        1  
5              Sanjay Leela Bhansali     0.0        6  
...                              ...     ...      ...  
1275                      V.K. Singh     0.0        1  
1276                     Raj Purohit     0.0        1  
1277                  Satish Rajwade     0.0        1  
1278                 Bhabhani Prasad     0.0        1  
1283  Rohit Malhotra | Anubhav Sinha     0.0        2  

[1131 rows x 10 columns]

Now the data has 1131 rows

Lets check to verify there are no null values left

In [67]:
print(actor.isna().sum())
print('****************************')
print(director.isna().sum())
print('****************************')
print(movie.isna().sum())

actorId                 0
actorName               0
movieCount              0
ratingSum               0
normalizedMovieRank     0
googleHits              0
normalizedGoogleRank    0
normalizedRating        0
dtype: int64
****************************
directorId              0
directorName            0
movieCount              0
ratingSum               0
normalizedMovieRank     0
googleHits              0
normalizedGoogleRank    0
normalizedRating        0
dtype: int64
****************************
imdbId         0
title          0
releaseYear    0
releaseDate    0
genre          0
writers        0
actors         0
directors      0
sequel         0
hitFlop        0
dtype: int64


Lets split the actor column in movie to get only the first and second actor name

In [68]:
movie['first_lead_actor'] = movie['actors'].str.split('|').str[0]
movie['second_lead_actor'] = movie['actors'].str.split('|').str[1]

In [69]:
movie

imdbId                              title  releaseYear releaseDate  \
0     tt0118578                             Albela         2001   20-Apr-01   
1     tt0169102  Lagaan: Once Upon a Time in India         2001   08-May-02   
3     tt0222024             Hum Tumhare Hain Sanam         2002   24-May-02   
4     tt0227194                         One 2 Ka 4         2001   30-Mar-01   
5     tt0238936                             Devdas         2002   12-Jul-02   
...         ...                                ...          ...         ...   
1275  tt3893476                              Spark         2014   10-Oct-14   
1276  tt4010306                          Jigariyaa         2014   10-Oct-14   
1277  tt4045666                Mumbai Delhi Mumbai         2014   05-Dec-14   
1278  tt4119196                  Love U Crazy Girl         2014   09-May-14   
1283  tt4228746                                Zid         2014   28-Nov-14   

                            genre  \
0                         Romance   
1     Adventure | Drama | Musical   
3                 Drama | Romance   
4         Action | Comedy | Drama   
5               Musical | Romance   
...                           ...   
1275                       Action   
1276                        Drama   
1277             Comedy | Romance   
1278                      Mystery   
1283                     Thriller   

                                                writers  \
0     Honey Irani (screenplay) | Honey Irani (story)...   
1     Ashutosh Gowariker (story) | Ashutosh Gowarike...   
3     K.S. Adiyaman | Arun Kumar (assistant dialogue...   
4     Sanjay Chhel | Raaj Kumar Dahima (screenplay) ...   
5     Saratchandra Chatterjee (based on the novel by...   
...                                                 ...   
1275                                 V.K. Singh (story)   
1276       Vinod Bachchan | Apratim Khare | Raj Purohit   
1277      Manu Rishi Chadda (dialogue) | Satish Rajwade   
1278                                    Bhabhani Prasad   
1283                   Vivek Agnihotri | Rohit Malhotra   

                                                 actors  \
0     Govinda | Aishwarya Rai Bachchan | Jackie Shro...   
1     Aamir Khan | Gracy Singh | Rachel Shelley | Pa...   
3     Shah Rukh Khan | Madhuri Dixit | Salman Khan |...   
4     Shah Rukh Khan | Juhi Chawla | Jackie Shroff |...   
5     Shah Rukh Khan | Madhuri Dixit | Aishwarya Rai...   
...                                                 ...   
1275  Rati Agnihotri | Mansha Bahl | Rajneesh Duggal...   
1276  Deepak Chadha | Harshvardhan Deo | Sneha Deori...   
1277                          Shiv Pandit | Piaa Bajpai   
1278  Mohan Joshi | Shakti Kapoor | Nitika Sharma | ...   
1283  Shradha Das | Mohan Kapoor | Mannara | Karanvi...   

                           directors  sequel  hitFlop first_lead_actor  \
0                      Deepak Sareen     0.0        2         Govinda    
1                 Ashutosh Gowariker     0.0        6      Aamir Khan    
3                      K.S. Adiyaman     0.0        4  Shah Rukh Khan    
4                  Shashilal K. Nair     0.0        1  Shah Rukh Khan    
5              Sanjay Leela Bhansali     0.0        6  Shah Rukh Khan    
...                              ...     ...      ...              ...   
1275                      V.K. Singh     0.0        1  Rati Agnihotri    
1276                     Raj Purohit     0.0        1   Deepak Chadha    
1277                  Satish Rajwade     0.0        1     Shiv Pandit    
1278                 Bhabhani Prasad     0.0        1     Mohan Joshi    
1283  Rohit Malhotra | Anubhav Sinha     0.0        2     Shradha Das    

             second_lead_actor  
0      Aishwarya Rai Bachchan   
1                 Gracy Singh   
3               Madhuri Dixit   
4                 Juhi Chawla   
5               Madhuri Dixit   
...                        ...  
1275              Mansha Bahl   
1276         Harshvardhan Deo   
1277       

In [74]:
csv_file_path = 'C:/Users/HP/Desktop/Data Sets/bollywood_movie/movie.csv'

In [76]:
movie.to_csv(csv_file_path, index=False)
print("CSV file saved successfully at:", csv_file_path)

CSV file saved successfully at: C:/Users/HP/Desktop/Data Sets/bollywood_movie/movie.csv


In [77]:
csv_file_path = 'C:/Users/HP/Desktop/Data Sets/bollywood_movie/actor.csv'

In [78]:
actor.to_csv(csv_file_path, index=False)
print("CSV file saved successfully at:", csv_file_path)

CSV file saved successfully at: C:/Users/HP/Desktop/Data Sets/bollywood_movie/actor.csv


In [79]:
csv_file_path = 'C:/Users/HP/Desktop/Data Sets/bollywood_movie/director.csv'

In [80]:
director.to_csv(csv_file_path, index=False)
print("CSV file saved successfully at:", csv_file_path)

CSV file saved successfully at: C:/Users/HP/Desktop/Data Sets/bollywood_movie/director.csv
